src/Domain/CoordinateMaps/Wedge.hpp

```diff
@@ -541,18 +801,30 @@ class Wedge {
    * Constructs a 3D wedge.
    * \param radius_inner Distance from the origin to one of the corners which
    * lie on the inner surface.
-   * \param radius_outer Distance from the origin to one of the corners which
-   * lie on the outer surface.
+   * \param radius_outer For any Wedge with zero `focal_offset` or a spherical
+   * Wedge with nonzero `focal_offset`, this is the distance from the
+   * `focal_offset` to one of the corners that lie on the outer surface. For a
+   * Wedge with both a nonzero `focal_offset` and `outer_sphericity == 0.0`,
+   * this parameter has no effect because the outer corners of the Wedge will
+   * instead lie on the parent surface, which is set by the `cube_half_length`
+   * (see Wedge docs for more details).
    * \param orientation_of_wedge The orientation of the desired wedge relative
    * to the orientation of the default wedge which is a wedge that has its
    * curved surfaces pierced by the upper-z axis. The logical $\xi$ and $\eta$
    * coordinates point in the cartesian x and y directions, respectively.
    * \param sphericity_inner Value between 0 and 1 which determines
    * whether the inner surface is flat (value of 0), spherical (value of 1) or
-   * somewhere in between.
+   * somewhere in between. If `focal_offset` is nonzero, `sphericity_inner` must
+   * be `1.0`.
    * \param sphericity_outer Value between 0 and 1 which determines
    * whether the outer surface is flat (value of 0), spherical (value of 1) or
-   * somewhere in between.
+   * somewhere in between. If `focal_offset` is nonzero, `sphericity_outer` must
+   * be `0.0` or `1.0`.
+   * \param cube_half_length Half the length of the parent surface (see Wedge
+   * documentation for more details). This parameter has no effect when
+   * `focal_offset` is zero.
```

Comment:
We usually don't like having parameters that are completely ignored based on some other parameter. It can make debugging hard if you have to track down a number just to realize it didn't matter in the first place. This seems like a good example to use a `std::optional` with  `std::pair` since the cube half length and focal offset are intimately tied. So the option in the constructor would be

```cpp
Wedge(...
      const std::optional<std::pair<std::array<double, 3>, double>>&
             focal_offset_and_half_cube_length,
      ...);
```

This way nothing is ignored and as a bonus, you can know exactly (not approximately) if you have an offset.

Reply:
Discussion on 8/21/24: The focal offset member should probably be a `std::optional`. This can be the sentinel for other things. The other variables that won't be used in one case vs the other can just be initialized to signaling NaN.